store_config_in_the_environment.md

```diff
@@ -4,9 +4,9 @@ Store the configuration in the environment
 General idea
 ------------
 
```

Comment:
Maybe add why it is harmful to commit secrets? Also how hard it is to remove and how you save yourself a lot of headaches by setting this up upfront? (although you do address it in the introduction)

Reply:
> Maybe add why it is harmful to commit secrets?

I asked myself if this was necessary and removed it for the sake of brevity. (I'm assuming that "e.g. secrets, should be always be kept secret" covers the topic.)

> [...] by setting this up upfront?

I have been actively trying to avoid prescribing doing things upfront _vs_ iterating because I think trying to _learen to set tings up properly_ is somethings that has prevented me to do anything at all for a long(er) time (that I wish it had).

> Also how hard it is to remove and how you save yourself a lot of headaches [...]

As for the first part, I also find rotating secrets annoying, but it has often felt more boring to me than actually difficult, so I'd rather not put too much emphasis on that.
In my opinion: if deploying your code meant getting some secrets in your repo, and you get to clean them up afterwards, and rotate them... and decide you'll spare yourself that work next time; then maybe the learnings were worth burning a couple of API keys once? YMMV, but I wouldn't make secrets management too much of a special case in the broader topic of configuration management.

Reply:
I guess it matters if they are your keys, or sensitive data that belongs to someone else... If it is your own data, than you can take more liberties I guess.
Thanks for the feedback! :)

Reply:
Yes, as always, context matters! :wink: